linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -575,54 +620,69 @@ private[emitter] class FunctionEmitter(jsGen: JSGen) {
                 transformExprNoChar(newValue))
           }
 
-        case While(cond, body, label) =>
+        case While(cond, body) =>
+          /* If there is a Labeled block immediately enclosed within the body
+           * of this while loop, acquire its label and use it as the while's
+           * label, turning it into a `continue` label.
+           */
+          val (optLabel, newBody) = {
+            body match {
+              case Labeled(label, _, innerBody) =>
+                val innerBodyEnv = env
+                  .withLabeledExprLHS(label, Lhs.Discard)
+                  .withTurnLabelIntoContinue(label.name)
+                  .withDefaultBreakTargets(tailPosLabels)
+                  .withDefaultContinueTargets(Set(label.name))
+                val newBody =
+                  pushLhsInto(Lhs.Discard, innerBody, Set(label.name))(innerBodyEnv)
+                val optLabel = if (usedLabels.contains(label.name))
+                  Some(transformLabelIdent(label))
+                else
+                  None
+                (optLabel, newBody)
+
+              case _ =>
+                val bodyEnv = env
+                  .withDefaultBreakTargets(tailPosLabels)
+                  .withDefaultContinueTargets(Set.empty)
+                val newBody = transformStat(body, Set.empty)(bodyEnv)
+                (None, newBody)
+            }
+          }
+
           /* We cannot simply unnest(cond) here, because that would eject the
            * evaluation of the condition out of the loop.
            */
-          val newLabel = label.map(transformLabelIdent)
-          val bodyBreakTargets = tailPosLabels ++ label.map(_.name)
           if (isExpression(cond)) {
-            js.While(transformExprNoChar(cond),
-                transformStat(body, Set.empty)(
-                    env.withDefaultBreakTargets(bodyBreakTargets)),
-                newLabel)
+            js.While(transformExprNoChar(cond), newBody)
           } else {
             js.While(js.BooleanLiteral(true), {
               unnest(cond) { (newCond, env0) =>
                 implicit val env = env0
-                js.If(transformExprNoChar(newCond),
-                    transformStat(body, Set.empty)(
-                        env.withDefaultBreakTargets(bodyBreakTargets)),
-                    js.Break())
+                js.If(transformExprNoChar(newCond), newBody, js.Break())
               }
-            }, newLabel)
+            })
```

Comment:
Don't you need to use `optLabel` here? Otherwise how will the while get a label if it needs one?

Reply:
Ouch, yes indeed!

I guess all the cases that are generated by Scala source code end up not needing a label ... which is why it wasn't caught by the CI. But in general, we might need one.